File: app/src/main/kotlin/com/openclaw/ghostcrab/domain/model/SkillInstallProgress.kt
package com.openclaw.ghostcrab.domain.model

sealed interface SkillInstallProgress {
    data object Idle : SkillInstallProgress
    data class Connecting(val target: String) : SkillInstallProgress
    data class Downloading(val pct: Int?) : SkillInstallProgress
    data class Verifying(val sha256Prefix: String) : SkillInstallProgress
    data class Applying(val step: String) : SkillInstallProgress
    data class Succeeded(val installed: InstalledSkill) : SkillInstallProgress
    data class Failed(val error: SkillInstallError) : SkillInstallProgress
}

File: ios/GhostCrab/Domain/Model/SkillInstallProgress.swift
import Foundation

/// Progress reporting for an in-app skill install operation.
public enum SkillInstallProgress: Sendable, Hashable {
    case idle
    case connecting(target: String)
    case downloading(pct: Int?)
    case verifying(sha256Prefix: String)
    case applying(step: String)
    case succeeded(installed: InstalledSkill)
    case failed(error: SkillInstallError)
}
